system/lib/libc/musl/src/thread/pthread_detach.c

```diff
@@ -3,10 +3,11 @@
 
 static int __pthread_detach(pthread_t t)
 {
-	// XXX EMSCRIPTEN: Add check for invalid (already joined) thread.  Again
-	// for the benefit of the conformance tests.
-	if (t->self != t)
-		return ESRCH;
+#ifdef __EMSCRIPTEN__
+	// Attempt to detach a thread which does not point to a valid thread, or
+	// does not exist anymore.
```

Comment:
Why not keep the old comment.. its seems more useful in that it says why we are doing it.  Is it not true?   

The new comment does say why emscripten needs this patch..

Reply:
The reason for the doc comment change was that the "(already joined)"-sentence wasn't quite right. The check is there only for invalid or non-existent threads (AFAIK).

I reverted it partially with commit 1e1761b09811a71c4edd85f956193999d1dc098c, so that it indicates why the patch is needed.

Reply:
Oh BTW, sometimes there are 2 spaces after a sentence, I've seen that a few times now in comments. Do I have to follow that convention?

Reply:
But I believe the check for validity is to protect specifically again detaching already joined thread (which the testsuite does IIRC).    So its probably worth continuing to mention why we might care specifically about a validity here, even though its undefined behaviour and we technically should not need to care.

Reply:
Ah, I understand now. Commit 7806443bae105e1fec61c0c4b1092aa686bbb3b8 brings the comment to its original state. I was misinterpreting that comment as "already-detached", which is checked below.
https://github.com/emscripten-core/emscripten/blob/1e1761b09811a71c4edd85f956193999d1dc098c/system/lib/libc/musl/src/thread/pthread_detach.c#L11-L12

Already joined threads are just threads that are invalid (non-existent).